Firebase Fix to "Firebase is not defined"

TomKwee opened the pull request:
I got a good chance to investigate this problem this week. The problem is due to the Firebase upgrade to 3.x. There is documentation at https://firebase.google.com/support/guides/firebase-web. The point is that code of new Firebase is outdated, simply use firebase.database().ref(). I only have to change map.js to fix that.

The strange thing is that I did not get the same error as yours until I changed the code referenced to www.gstatic.com/firebasejs/3.5.1. So my first instinct is wrong, about not having the proper reference to that link.
I am using a different way to add/commit code changes. I used the web Github, again, and this time, I commit changes from the <>Code tab. I uploaded all 5 files for your convenience.

I was hoping to make this code fix earlier. Eager to see the change now :-)

TomKwee committed:
Firebase Fix to "Firebase is not defined"

I got a good chance to investigate this problem this week. The problem is due to the Firebase upgrade to 3.x. There is documentation at https://firebase.google.com/support/guides/firebase-web. The point is that code of new Firebase is outdated, simply use firebase.database().ref(). I only have to change map.js to fix that.

The strange thing is that I did not get the same error as yours until I changed the code referenced to www.gstatic.com/firebasejs/3.5.1. So my first instinct is wrong, about not having the proper reference to that link.
I am using a different way to add/commit code changes. I used the web Github, again, and this time, I commit changes from the <>Code tab. I uploaded all 5 files for your convenience.

I was hoping to make this code fix earlier. Eager to see the change now :-)

olenakozdoba commented:
Tommy, I switched to the Tommy_branch and applied the fix for Firebase (firebase.database().ref() instead of new Firebase("..."))=> the JS error is gone, but Reserve/Release functionality isn't working. No errors, just after page refresh it's not changing the pin. Is this working for you?

TomKwee commented:
It is working for me. I tested the app on Github, https://easychargeapp.firebaseapp.com, and it has the same problem that you mentioned. The app simply could not change the "status" field onto the Firebase. My Chrome dev tool reports error of "Synchronous XMLHttpRequest on the main thread is deprecated..." on code addressRef.update(). However I do not get that error on my local environment. So it's a puzzlement.

I'll look into it further. I think it was working before even for the app on Github, **right?**
It seems the problem is related to Firebase and my local environment is different but I do not know what that can be for now.

TomKwee committed:
Add files via upload

I made small minor code fixes. app.js (fix bottom-container issue) and map.js (reporting error from calling update()).

Section 1:
Before I mention the possible solution, let’s try to see the run-time error. The problem with the reserve/release feature is with the call addressRef.update() in map.js. As I said, I see this error on the existing app released from Github (https://easychargeapp.firebaseapp.com). 

The error that I only see was “Synchronous XMLHttpRequest on the main thread…”. However database updates are naturally asynchronous, not synchronous as the error says.  So if you get an error from the update() call, you would not see it immediately since it is asynchronous. And worse, with the Chrome devtool, the error is actually erased from the log when the web page is refreshed (after the update() call), hence I do not see the error; perhaps that’s what you experienced. You can see the error if you set the breakpoint on the right place. In the following, here are m
